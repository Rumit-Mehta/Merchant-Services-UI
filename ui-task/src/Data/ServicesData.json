[
  {
    "id": 1,
    "service": "Service 1",
    "description": "Description 1",
    "isChecked": false
  },
  {
    "id": 2,
    "service": "Service 2",
    "description": "Description 2",
    "isChecked": false
  },
  {
    "id": 3,
    "service": "Service 3",
    "description": "Description 3",
    "isChecked": false
  },
  {
    "id": 4,
    "service": "Service 4",
    "description": "Description 4",
    "isChecked": false
  }
]
