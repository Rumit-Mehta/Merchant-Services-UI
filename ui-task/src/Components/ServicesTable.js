import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Checkbox from "@material-ui/core/Checkbox";
import ServicesData from "../Data/ServicesData.json";
import TableContainer from "@material-ui/core/TableContainer";
import Paper from "@material-ui/core/Paper";

let newSelected = [];
let maxServices = 0;

ServicesData.map((element) => {
  maxServices++;
});

const StyledTableCell = withStyles((theme) => ({
  head: {
    color: theme.palette.primary.main,
    fontWeight: "bold",
  },
  body: {
    color: theme.palette.secondary.main,
  },
}))(TableCell);

function EnhancedTableHead(props) {
  const { onSelectAllClick, numSelected, rowCount } = props;

  return (
    <TableHead>
      <TableRow>
        <StyledTableCell padding="checkbox">
          <Checkbox
            color="primary"
            indeterminate={numSelected > 0 && numSelected < rowCount}
            checked={rowCount > 0 && numSelected === rowCount}
            onChange={onSelectAllClick}
          />
        </StyledTableCell>
        <StyledTableCell align={"left"} padding={"default"}>
          Service
        </StyledTableCell>
        <StyledTableCell align={"left"} padding={"default"}>
          Description
        </StyledTableCell>
      </TableRow>
    </TableHead>
  );
}

export default function EnhancedTable() {
  const [selected, setSelected] = React.useState([]);
  const [page] = React.useState(0);
  const [dense] = React.useState(false);
  const [rowsPerPage] = React.useState(5);

  const handleSelectAllClick = (event) => {
    if (event.target.checked) {
      const newSelecteds = ServicesData.map((service) => service.name);
      newSelected.length = maxServices;
      console.log("im here");
      setSelected(newSelecteds);
      return;
    } else {
      newSelected.length = 0;
    }
    setSelected([]);
  };

  const handleClick = (event, name) => {
    newSelected = [];
    const selectedIndex = selected.indexOf(name);
    if (selectedIndex === -1) {
      newSelected = newSelected.concat(selected, name);
    } else if (selectedIndex === 0) {
      newSelected = newSelected.concat(selected.slice(1));
    } else if (selectedIndex === selected.length - 1) {
      newSelected = newSelected.concat(selected.slice(0, -1));
    } else if (selectedIndex > 0) {
      newSelected = newSelected.concat(
        selected.slice(0, selectedIndex),
        selected.slice(selectedIndex + 1)
      );
    }
    console.log(newSelected);
    setSelected(newSelected);
  };

  const isSelected = (name) => selected.indexOf(name) !== -1;

  const emptyRows =
    rowsPerPage -
    Math.min(rowsPerPage, ServicesData.length - page * rowsPerPage);

  return (
    <div>
      <Paper elevation={0} square={false}>
        <TableContainer>
          <Table
            className="ServiceTable"
            aria-labelledby="tableTitle"
            size={dense ? "small" : "medium"}
            aria-label="enhanced table"
          >
            <EnhancedTableHead
              numSelected={selected.length}
              onSelectAllClick={handleSelectAllClick}
              rowCount={ServicesData.length}
            />
            <TableBody>
              {ServicesData.map((row, index) => {
                const isItemSelected = isSelected(row.name);
                const labelId = `enhanced-table-checkbox-${index}`;

                return (
                  <TableRow
                    key={row.id}
                    hover
                    onClick={(event) => handleClick(event, row.name)}
                    role="checkbox"
                    aria-checked={isItemSelected}
                    selected={isItemSelected}
                  >
                    <StyledTableCell padding="checkbox">
                      <Checkbox
                        color="primary"
                        checked={isItemSelected}
                        inputProps={{ "aria-labelledby": labelId }}
                      />
                    </StyledTableCell>
                    <StyledTableCell id={labelId} padding="default">
                      {row.name}
                    </StyledTableCell>
                    <TableCell align="left">{row.description}</TableCell>
                  </TableRow>
                );
              })}
              {emptyRows > 0 && (
                <TableRow style={{ height: (dense ? 33 : 53) * emptyRows }}>
                  <TableCell colSpan={6} />
                </TableRow>
              )}
            </TableBody>
          </Table>
        </TableContainer>
      </Paper>
    </div>
  );
}
export { newSelected };
