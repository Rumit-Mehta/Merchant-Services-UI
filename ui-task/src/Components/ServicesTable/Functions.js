import React from "react";
import ServicesData from "../../Data/ServicesData.json";

// Initialize variables
const maxServices = ServicesData.length;
let newSelected = [];

export default () => {
  // This code handles the selection of each checkbox
  const handleClick = (event, name) => {
    newSelected = [];
    const selectedIndex = selected.indexOf(name);
    if (selectedIndex === -1) {
      newSelected = newSelected.concat(selected, name);
    } else if (selectedIndex === 0) {
      newSelected = newSelected.concat(selected.slice(1));
    } else if (selectedIndex === selected.length - 1) {
      newSelected = newSelected.concat(selected.slice(0, -1));
    } else if (selectedIndex > 0) {
      newSelected = newSelected.concat(
        selected.slice(0, selectedIndex),
        selected.slice(selectedIndex + 1)
      );
    }

    setSelected(newSelected);
  };

  // function that deals with what happens when the "select all" checkbox is ticked
  // If not already selected: it changes length of selected service to max amount of services and selects all services.
  // If already selected: it changes length of selected service to 0 and un-selects the services
  const handleSelectAllClick = (event) => {
    if (event.target.checked) {
      const newSelecteds = ServicesData.map((service) => service.name);
      newSelected.length = maxServices;
      setSelected(newSelecteds);
      return;
    } else {
      newSelected.length = 0;
    }
    setSelected([]);
  };

  const [selected, setSelected] = React.useState([]);
  const [page] = React.useState(0);
  const [dense] = React.useState(false);
  const [rowsPerPage] = React.useState(5);
  const isSelected = (name) => selected.indexOf(name) !== -1;

  // const to find the amount of empty rows in the table
  const emptyRows =
    rowsPerPage -
    Math.min(rowsPerPage, ServicesData.length - page * rowsPerPage);

  return {
    newSelected,
    handleClick,
    emptyRows,
    handleSelectAllClick,
    selected,
    setSelected,
    page,
    dense,
    rowsPerPage,
    isSelected,
  };
};
export { newSelected };
