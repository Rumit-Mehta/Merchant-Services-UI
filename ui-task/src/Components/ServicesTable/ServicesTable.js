import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Checkbox from "@material-ui/core/Checkbox";
import ServicesData from "../../Data/ServicesData.json";
import TableContainer from "@material-ui/core/TableContainer";
import Paper from "@material-ui/core/Paper";

// Initialize variables
let newSelected = [];
let maxServices = 0;

// Find the max services available from the ServicesData.json file
ServicesData.map((element) => {
  maxServices++;
});

// Overiding 'TableCell' style by making the font bold and changing the colour.
const StyledTableCell = withStyles((theme) => ({
  head: {
    color: theme.palette.primary.main,
    fontWeight: "bold",
  },
  body: {
    color: theme.palette.secondary.main,
  },
}))(TableCell);

// Function which creates the table headers. 3 columns: checkbox, Service, Description
function EnhancedTableHead(props) {
  const { onSelectAllClick, numSelected, rowCount } = props;

  return (
    <TableHead>
      <TableRow>
        <StyledTableCell padding="checkbox" className="column">
          <Checkbox
            color="primary"
            indeterminate={numSelected > 0 && numSelected < rowCount}
            checked={rowCount > 0 && numSelected === rowCount}
            onChange={onSelectAllClick}
          />
        </StyledTableCell>
        <StyledTableCell align={"left"} padding={"default"} className="column">
          Service
        </StyledTableCell>
        <StyledTableCell align={"left"} padding={"default"} className="column">
          Description
        </StyledTableCell>
      </TableRow>
    </TableHead>
  );
}

// Default function which generates the body of the table as well handle to checkboxes
export default function EnhancedTable() {
  const [selected, setSelected] = React.useState([]);
  const [page] = React.useState(0);
  const [dense] = React.useState(false);
  const [rowsPerPage] = React.useState(5);

  // function that deals with what happens when the "select all" checkbox is ticked
  // If not already selected: it changes length of selected service to max amount of services and selects all services.
  // If already selected: it changes length of selected service to 0 and un-selects the services
  const handleSelectAllClick = (event) => {
    if (event.target.checked) {
      const newSelecteds = ServicesData.map((service) => service.name);
      newSelected.length = maxServices;
      setSelected(newSelecteds);
      return;
    } else {
      newSelected.length = 0;
    }
    setSelected([]);
  };

  // This code handles the selection of each checkbox
  const handleClick = (event, name) => {
    newSelected = [];
    const selectedIndex = selected.indexOf(name);
    if (selectedIndex === -1) {
      newSelected = newSelected.concat(selected, name);
    } else if (selectedIndex === 0) {
      newSelected = newSelected.concat(selected.slice(1));
    } else if (selectedIndex === selected.length - 1) {
      newSelected = newSelected.concat(selected.slice(0, -1));
    } else if (selectedIndex > 0) {
      newSelected = newSelected.concat(
        selected.slice(0, selectedIndex),
        selected.slice(selectedIndex + 1)
      );
    }
    console.log(newSelected);
    setSelected(newSelected);
  };

  const isSelected = (name) => selected.indexOf(name) !== -1;

  // const to find the amount of empty rows in the table
  const emptyRows =
    rowsPerPage -
    Math.min(rowsPerPage, ServicesData.length - page * rowsPerPage);

  // This is the main body of the table, and goes through the json file to add each service onto the table
  return (
    <div>
      <Paper elevation={0} square={false}>
        <TableContainer>
          <Table
            className="ServiceTable"
            aria-labelledby="tableTitle"
            size={dense ? "small" : "medium"}
            aria-label="enhanced table"
          >
            <EnhancedTableHead
              numSelected={selected.length}
              onSelectAllClick={handleSelectAllClick}
              rowCount={ServicesData.length}
            />
            <TableBody>
              {ServicesData.map((row, index) => {
                const isItemSelected = isSelected(row.name);
                const labelId = `enhanced-table-checkbox-${index}`;

                return (
                  <TableRow
                    key={row.id}
                    hover
                    onClick={(event) => handleClick(event, row.name)}
                    role="checkbox"
                    aria-checked={isItemSelected}
                    selected={isItemSelected}
                  >
                    <StyledTableCell padding="checkbox">
                      <Checkbox
                        color="primary"
                        checked={isItemSelected}
                        inputProps={{ "aria-labelledby": labelId }}
                      />
                    </StyledTableCell>
                    <StyledTableCell id={labelId} padding="default">
                      {row.name}
                    </StyledTableCell>
                    <TableCell align="left">{row.description}</TableCell>
                  </TableRow>
                );
              })}
              {emptyRows > 0 && (
                <TableRow style={{ height: (dense ? 33 : 53) * emptyRows }}>
                  <TableCell colSpan={6} />
                </TableRow>
              )}
            </TableBody>
          </Table>
        </TableContainer>
      </Paper>
    </div>
  );
}
export { newSelected, StyledTableCell };
