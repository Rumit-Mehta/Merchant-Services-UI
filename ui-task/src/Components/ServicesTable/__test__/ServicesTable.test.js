import React from "react";
import ReactDOM from "react-dom";
import ServicesTable from "../ServicesTable";
import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { TableRow } from "@material-ui/core";
import ServicesData from "../../../Data/ServicesData.json";
import { StyledTableCell } from "../ServicesTable";

afterEach(cleanup);
test("renders  without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ServicesTable />, div);
});

test("table has correct rows and columns", () => {
  let maxServices = 0;

  ServicesData.map((element) => {
    maxServices++;
  });

  const wrapper = shallow(<ServicesTable />);
  const rows = wrapper.find(TableRow);
  expect(rows).toHaveLength(Number(maxServices));

  const dataEntries = wrapper.find(StyledTableCell);
  const columns = 2;
  expect(dataEntries.length / rows.length).toEqual(columns);
});

test("matches snapshot", () => {
  const tree = renderer.create(<ServicesTable />).toJSON();
  expect(tree).toMatchSnapshot();
});
