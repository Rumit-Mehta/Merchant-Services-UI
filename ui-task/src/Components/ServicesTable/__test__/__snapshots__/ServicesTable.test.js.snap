// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
  >
    <div
      className="MuiTableContainer-root"
    >
      <table
        aria-label="enhanced table"
        aria-labelledby="tableTitle"
        className="MuiTable-root ServiceTable"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 column MuiTableCell-paddingCheckbox"
              scope="col"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-6"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 column MuiTableCell-alignLeft"
              scope="col"
            >
              Service
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head WithStyles(ForwardRef(TableCell))-head-1 column MuiTableCell-alignLeft"
              scope="col"
            >
              Description
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            aria-checked={false}
            className="MuiTableRow-root MuiTableRow-hover"
            onClick={[Function]}
            role="checkbox"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-0"
                    checked={false}
                    className="PrivateSwitchBase-input-6"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2"
              id="enhanced-table-checkbox-0"
            >
               Consulting
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
            </td>
          </tr>
          <tr
            aria-checked={false}
            className="MuiTableRow-root MuiTableRow-hover"
            onClick={[Function]}
            role="checkbox"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-1"
                    checked={false}
                    className="PrivateSwitchBase-input-6"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2"
              id="enhanced-table-checkbox-1"
            >
               Adivce
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
            </td>
          </tr>
          <tr
            aria-checked={false}
            className="MuiTableRow-root MuiTableRow-hover"
            onClick={[Function]}
            role="checkbox"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-2"
                    checked={false}
                    className="PrivateSwitchBase-input-6"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2"
              id="enhanced-table-checkbox-2"
            >
               Billing
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
            </td>
          </tr>
          <tr
            aria-checked={false}
            className="MuiTableRow-root MuiTableRow-hover"
            onClick={[Function]}
            role="checkbox"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-3"
                    checked={false}
                    className="PrivateSwitchBase-input-6"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2"
              id="enhanced-table-checkbox-3"
            >
               Disputes
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
            </td>
          </tr>
          <tr
            aria-checked={false}
            className="MuiTableRow-root MuiTableRow-hover"
            onClick={[Function]}
            role="checkbox"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2 MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    aria-labelledby="enhanced-table-checkbox-4"
                    checked={false}
                    className="PrivateSwitchBase-input-6"
                    data-indeterminate={false}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-2"
              id="enhanced-table-checkbox-4"
            >
               Transactions
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore 
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
